docs/citus.rst

```diff
@@ -0,0 +1,334 @@
+
+__ https://docs.citusdata.com/en/stable/installation/multi_node.html
+
+TL;DR
+-----
+
+There are only a few simple rules you need to follow:
+
+1. Citus extension must be available on all nodes.
+2. Cluster name (``scope``) must be the same for all Citus nodes!
+3. Superuser credentials must be the same on coordinator and all worker
+   nodes, and ``pg_hba.conf`` should allow superuser access between all nodes.
+4. :ref:`REST API <restapi_settings>` access should be allowed from worker
+   nodes to the coordinator. E.g., credentials should be the same and if
+   configured, client certificates from worker nodes must be accepted by the
+   coordinator.
+5. Add the following section to the ``patroni.yaml``:
+
+.. code:: YAML
+
+        citus:
+          group: X  # 0 for coordinator and 1, 2, 3, etc for workers
+          database: citus  # must be the same on all nodes
+
+
+After that you just need to start Patroni and it will handle the rest:
+
+1. ``citus`` extension will be automatically added to ``shared_preload_libraries``.
+2. If ``max_prepared_transactions`` isn't explicitly set in the global
+   :ref:`dynamic configuration <dynamic_configuration>` Patroni will
+   automatically set it to ``2*max_connections``.
+3. The ``citus.database`` will be automatically created followed by ``CREATE EXTENSION citus``.
+4. Current superuser :ref:`credentials <postgresql_settings>` will be added to the ``pg_dist_authinfo``
+   table to allow cross-node communication. Don't forget to update them if
+   later you decide to change superuser username/password/sslcert/sslkey!
+5. The coordinator primary node will automatically discover worker primary
+   nodes and add them to the ``pg_dist_node`` table using the
+   ``citus_add_node()`` function.
+6. Patroni will also maintain ``pg_dist_node`` in case failover/switchover
+   on the coordinator or worker clusters occurs.
+
+patronictl
+----------
+
+Coordinator and worker clusters are physically different PostgreSQL/Patroni
+clusters that are just logically groupped together using Citus. Therefore in
+most cases it is not possible to manage them as a single entity.
+
+It results in two major differences in ``patronictl`` behaviour when
+``patroni.yaml`` has the ``citus`` section comparing with the usual:
+
+1. The ``list`` and the ``topology`` by default output all members of the Citus
+   formation (coordinators and workers). The new column ``Group`` indicates
+   which Citus group they belong to.
+2. For all ``patronictl`` commands the new option is introduced, named
+   ``--group``. For some commands the default value for the group might be
+   taken from the ``patroni.yaml``. For example, ``patronictl pause`` will
+   enable the maintenance mode by default for the ``group`` that is set in the
+   ``citus`` section, but for example for ``patronictl  switchover`` or
+   ``patronictl remove`` the group must be explicitly specified.
```

Comment:
What's the reasoning for this difference in behavior with regards to patroni.yaml between commands?

Reply:
It is a long story.
Originally every command required explicit cluster name to be specified in the command-line, because it was supposed to work for an arbitrary Patroni clusters registered in DCS.

But, it turned out that people mostly run `patronictl` on the same host where Patroni runs.
Providing the cluster name was very annoying (especially knowing that it is already in the config file), therefore for most commands we made the cluster name optional. If the cluster name isn't supplied, it is picked from the config file.
Known exceptions to this rule are:
1. commands which are followed by a member name, e.g.: `patronictl reinit $CLUSTER_NAME $MEMBER_NAME`.
2. "dangerous" command: `patronictl remove $CLUSTER_NAME`. I call in dangerous because it wipes all keys for a given cluster from DCS.

Since a Citus cluster is effectively just a bunch of different Patroni clusters logically connected using the groupid I had to introduce the `--group` option for all commands. And now we have more or less the same story.
1. For `patronictl list` we probably don't want to use the default group value from the config because typically users want to see all coordinator and worker nodes as a whole.
2. `patronictl remove` command still remains dangerous, and since we require the cluster name to be explicitly specified we have to do the same for the `--group`. Actually, the `remove` command has only an interactive mode and asks to confirm the cluster name. Probably we should do the same for the group.
3. `patronictl switchover/failover` -- the workflow of these commands implies that it works only for a single Patroni cluster, therefore the group must be specified either explicitly from the command line, or it will ask for while working in the interactive mode.

For remaining commands the workflow will stay as it is. Lets say you investigate the problem with one of the workers. In this case most likely you will be on the host where the worker is running and want `patronictl` do some actions (edit-config/show-config/pause/resume) for a given group.